### PR TITLE
272670_Caching-No-Content-Fixe

### PR DESCRIPTION
**Pull request by fshehzad-defra**
272670, Handle empty/no content as success

### CODE REVIEW

File: EST.MIT.Invoice.Api/Services/Api/CachedReferenceDataApi.cs [hunk 1]
```diff
@@ -92,12 +92,22 @@ private async Task<ApiResponse<IEnumerable<CombinationForRoute>>> GetCombination
 
         _logger.LogInformation($"Calling Reference Data API for Route Combinations");
 
-        if (response.StatusCode == HttpStatusCode.OK)
+        if (response.StatusCode == HttpStatusCode.NoContent)
+        {
+            return new ApiResponse<IEnumerable<CombinationForRoute>>(HttpStatusCode.OK)
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api/Services/Api/CachedReferenceDataApi.cs:97]: I think it's OK to return HttpStatusCode.NoContent, as that's 204 which is a success code.

File: EST.MIT.Invoice.Api.Test/Services/Api/CachedReferenceDataApiService/GetRouteCombinationsAsyncTests.cs [hunk 2]
```diff
@@ -239,7 +239,7 @@ public async Task GetCombinationsListForRouteAsync_ResponseDataTaskIsFaulted_Log
         }
 
         [Fact]
-        public async Task GetCombinationsListForRouteAsync_ResponseDataIsNull_ReturnsNotFound()
+        public async Task GetCombinationsListForRouteAsync_ResponseDataIsNull_Returns_OK_And_Empty_List()
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api.Test/Services/Api/CachedReferenceDataApiService/GetRouteCombinationsAsyncTests.cs:242]: Is this intended to handle the 204 response?

File: EST.MIT.Invoice.Api/Services/Api/CachedReferenceDataApi.cs [hunk 3]
```diff
@@ -92,12 +92,22 @@ private async Task<ApiResponse<IEnumerable<CombinationForRoute>>> GetCombination
 
         _logger.LogInformation($"Calling Reference Data API for Route Combinations");
 
-        if (response.StatusCode == HttpStatusCode.OK)
+        if (response.StatusCode == HttpStatusCode.NoContent)
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api/Services/Api/CachedReferenceDataApi.cs:95]: Is there a test that covers this HttpStatusCode.NoContent handling?

File: EST.MIT.Invoice.Api/Services/Api/CachedReferenceDataApi.cs (hunk 1 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api/Services/Api/CachedReferenceDataApi.cs:97]: No its not. The consumer of this call is looking to check is the call was successful using the IsSuccess property on the APIResponse object. The IsSuccess is only true if is the status is OK or Created.

File: EST.MIT.Invoice.Api/Services/Api/CachedReferenceDataApi.cs (hunk 3 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api/Services/Api/CachedReferenceDataApi.cs:95]: Yes there is, I have added a test which tests that a NoContent returned from the repo gets handled and converts it into a OK and an empty payload as per the direction you suggested in our DM's.

**Review by andygibbonsdefra — APPROVED** [review]

**Pull request merged**